daemon/containerd/image_builder.go

```diff
@@ -524,18 +524,11 @@ func (i *ImageService) createImageOCI(ctx context.Context, imgToCreate imagespec
 		img.Labels[imageLabelClassicBuilderFromScratch] = "1"
 	}
 
-	createdImage, err := i.images.Update(ctx, img)
```

Comment:
This was calling `Update` specifically to handle the case where the image already exists - if update fails due to non found error, it would do the `Create` instead.

The new function handles the case a bit differently - instead of just updating the image, it will first remove it if it exists (or rename to a dangling reference if it's the last tag to that image).